c/tree.c

```diff
@@ -1145,8 +1153,15 @@ LOCAL void encode_node(MMDBW_tree_s *tree, MMDBW_node_s *node,
 
     uint32_t left =
         htonl(record_value_as_number(tree, &(node->left_record), args));
+    if (0 == left) {
```

Comment:
Couldn't we at least check that for normal nodes, it is larger? It looks like it should be easy to do.